docs/table/construct_table.rst

```diff
@@ -960,3 +964,71 @@ a `~astropy.table.Column` object with a ``unit`` attribute::
 
 To learn more about using standard `~astropy.table.Column` objects with defined
 units, see the :ref:`columns_with_units` section.
+
+
+Compatible tables
+^^^^^^^^^^^^^^^^^
+
+In order to improve interoperability between different table classes, an
+astropy |Table| object can be created directly from any other table
+object ``data`` that provides an ``__astropy_table__`` method.  In this case the
+``__astropy_table__`` method will be called as follows::
+
+  table = data.__astropy_table__(cls, copy)
+
+where ``cls`` is the |Table| class or subclass that is being instantiated,
+and ``copy`` indicates whether a copy of the values in ``data`` should be
+provided.  If ``copy`` is ``False`` then the ``__astropy_table__`` method
+must ensure that the column data can actually be passed by reference
+into an astropy |Table|.  If not then an exception must be raised.  This
+is the case if the source ``data`` are not in an object that can be passed
+by reference into a numpy ``np.ndarray`` subclass, for instance a pure Python
+list.
+
+As a simple example, imagine a dict-based table class.  (Note that |Table|
+already can be initialized from a dict-like object, so this is a bit contrived
+but does illustrate the principles involved.)  Please note the method
```

Comment:
(wordsmithing): double-"note" here, maybe "Pay attention to the method signature"?